packages/test-runner/src/testCollector.ts

```diff
@@ -18,50 +18,39 @@ import crypto from 'crypto';
 import { registrations, fixturesForCallback, rerunRegistrations } from './fixtures';
 import { Test, Suite, serializeConfiguration } from './test';
 import { RunnerConfig } from './runnerConfig';
-
+import { ReportFormat } from './reporters/json';
 
 export type Matrix = {
   [key: string]: string[]
 };
 
-export class TestCollector {
-  suite: Suite;
-
-  private _matrix: Matrix;
-  private _config: RunnerConfig;
-  private _grep: RegExp;
-  private _hasOnly: boolean;
-
-  constructor(suites: Suite[], matrix: Matrix, config: RunnerConfig) {
-    this._matrix = matrix;
-    this._config = config;
-    this.suite = new Suite('');
-    if (config.grep) {
-      const match = config.grep.match(/^\/(.*)\/(g|i|)$|.*/);
-      this._grep = new RegExp(match[1] || match[0], match[2]);
-    }
-    for (const suite of suites)
-      this._processSuite(suite);
-    this._hasOnly = this._filterOnly(this.suite);
-  }
-
-  hasOnly() {
-    return this._hasOnly;
+export function parseTests(suites: Suite[], matrix: Matrix, config: RunnerConfig): {suite?: Suite, parseError?: ReportFormat['parseError']} {
```

Comment:
Looks like a refactoring for the sake of refactoring, was it for better code understanding? If yes, let's not commit?

Reply:
It was needed to prove that I could return a parseError. Things get way uglier if it stays a class with fake state.